src/login/adal/DeviceTokenCredentials2.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, TokenCredential } from "@azure/core-auth";
+import { Constants as MSRestConstants, WebResource } from "@azure/ms-rest-js";
+import { DeviceTokenCredentials } from '@azure/ms-rest-nodeauth';
+import { TokenResponse } from "adal-node";
+
+/**
+ * Token that is forward compatible with track 2 Azure SDK for Node.js
+ * `getToken: Promise<AccessToken>` is required for use with T2 Azure SDK, but doesn't
+ * affect T1 SDKs as those require `signRequest`
+ * `DeviceTokenCredentials` forces `getToken` to return `TokenResponse` so this is to
+ * overwrite that implementation
+ */
+export class DeviceTokenCredentials2 extends DeviceTokenCredentials implements TokenCredential {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public async getToken(): Promise<any> {
+		const tokenResponse = <AccessToken & TokenResponse>(await this.getTokenFromCache(this.username));
+		return Object.assign(tokenResponse, {token: tokenResponse.accessToken, expiresOnTimestamp: tokenResponse.expiresIn});
```

Comment:
We could use `tokenResponse.expiresOn` here but that's a `Date | string` type so it seems more straightforward to do it this way

```suggestion
		return Object.assign(tokenResponse, {
			token: tokenResponse.accessToken, 
			expiresOnTimestamp: new Date().getTime() + tokenResponse.expiresIn 
		});
```

Reply:
Good catch.  I misunderstood `expiresIn` to be the time it expired in as a ms number.